### PR TITLE
feat!(node): add a method to get all blobs using shwap

### PR DESCRIPTION
**Pull request by zvolin**

### CODE REVIEW

File: ci/run-bridge.sh [hunk 1]
```diff
@@ -47,6 +47,15 @@ add_trusted_genesis() {
   sed -i'.bak' "s/TrustedHash = .*/TrustedHash = $genesis_hash/" "$CONFIG_DIR/config.toml"
 }
 
+whitelist_localhost_nodes() {
+  # to get the list of ips:
+  # cargo run -- node -n private -l 0.0.0.0
+  # docker compose -f ci/docker-compose.yml exec bridge-0 celestia p2p peer-info $lumina_peerid
+  dasel put -f "$CONFIG_DIR/config.toml" \
+    -t json -v '["172.18.0.1/24", "172.17.0.1/24", "192.168.0.186/24"]' \
```
Comment (oblique) [inline ci/run-bridge.sh:55]:
```suggestion
    -t json -v '["172.18.0.1/24", "172.17.0.1/24", "192.168.0.0/16"]' \
```

File: ci/run-bridge.sh [hunk 2]
```diff
@@ -9,7 +9,7 @@ NODE_NAME="bridge-$NODE_ID"
 # a private local network
 P2P_NETWORK="private"
 # a bridge node configuration directory
-CONFIG_DIR="$CELESTIA_HOME/.celestia-bridge-$P2P_NETWORK"
+CONFIG_DIR="$HOME/.celestia-bridge-$P2P_NETWORK"
```
Comment (fl0rek) [inline ci/run-bridge.sh:12]: why this change? :thinking:

File: node/src/node.rs [hunk 3]
```diff
@@ -371,13 +377,25 @@ where
...
     ) -> Result<RowNamespaceData> {
         Ok(self
             .p2p()
-            .get_row_namespace_data(namespace, row_index, block_height)
+            .get_row_namespace_data(namespace, row_index, block_height, timeout)
             .await?)
     }
 
+    /// Request all blobs with provided namespace in block corresponding to this header
+    /// on bitswap protocol.
```
Comment (fl0rek) [inline node/src/node.rs:389]:
```suggestion
    /// Request all blobs with provided namespace in the block corresponding to this header
    /// using bitswap protocol.
```

File: node/src/p2p.rs [hunk 4]
```diff
@@ -549,18 +554,65 @@ impl P2p {
...
-        // TODO: add timeout
-        let data = self.get_shwap_cid(cid, None).await?;
+        let data = self.get_shwap_cid(cid, timeout).await?;
         let row_namespace_data =
             RowNamespaceData::decode(id, &data[..]).map_err(|e| P2pError::Shwap(e.to_string()))?;
         Ok(row_namespace_data)
     }
 
+    /// Request all blobs with provided namespace in block corresponding to this header
+    /// on bitswap protocol.
```
Comment (fl0rek) [inline node/src/p2p.rs:570]: same as above 
```suggestion
    /// Request all blobs with provided namespace in the block corresponding to this header
    /// using bitswap protocol.
```

File: ci/run-bridge.sh (hunk 2 above)
Comment (zvolin) [reply ci/run-bridge.sh:12]: `CELESTIA_HOME` behavior [changed](https://github.com/celestiaorg/celestia-node/blob/main/nodebuilder/store.go#L189), it must now point to the exact directory where node will have it's config. In docker image we were setting it to just `HOME`. If you leave it unset, node will still set up using the previous path schema tho :thinking: not sure about reasoning, but there was something in release notes

Comment (fl0rek) [reply ci/run-bridge.sh:12]: I see, thank you :raised_hands:

**Review by fl0rek — APPROVED** [review]
LGTM!

**Review by fl0rek — APPROVED** [review]

**Review by oblique — APPROVED** [review]
